apps/whale-api/src/module.api/governance.controller.e2e.ts

```diff
@@ -0,0 +1,328 @@
+    await testing.container.waitForWalletBalanceGTE(100)
+    await testing.container.call('setgov', [
+      { ATTRIBUTES: { 'v0/params/feature/gov': 'true' } }
+    ])
+    await testing.container.generate(1)
+    app = await createTestingApp(container)
+    controller = app.get(GovernanceController)
+
+    /**
+     * Import the private keys of the masternode_operator in order to be able to mint blocks and vote on proposals.
+     * This setup uses the default masternode + two additional masternodes for a total of 3 masternodes.
+     */
+    await testing.rpc.wallet.importPrivKey(RegTestFoundationKeys[1].owner.privKey)
+    await testing.rpc.wallet.importPrivKey(RegTestFoundationKeys[1].operator.privKey)
+    await testing.rpc.wallet.importPrivKey(RegTestFoundationKeys[2].owner.privKey)
+    await testing.rpc.wallet.importPrivKey(RegTestFoundationKeys[2].operator.privKey)
+
+    // Create 1 CFP + 1 VOC
+    payoutAddress = await testing.generateAddress()
+    cfpProposalId = await testing.rpc.governance.createGovCfp({
+      title: 'CFP proposal',
+      context: 'github',
+      amount: new BigNumber(1.23),
+      payoutAddress: payoutAddress,
+      cycles: 2
+    })
+    await testing.container.generate(1)
+
+    vocProposalId = await testing.rpc.governance.createGovVoc({
+      title: 'VOC proposal',
+      context: 'github'
+    })
+    await testing.container.generate(1)
+
+    // Vote on CFP
+    await testing.rpc.governance.voteGov({
+      proposalId: cfpProposalId,
+      masternodeId: await getVotableMasternodeId(),
+      decision: VoteDecision.YES
+    })
+    await testing.container.generate(1)
+
+    // Expires cycle 1
+    const creationHeight = await testing.rpc.governance.getGovProposal(cfpProposalId).then(proposal => proposal.creationHeight)
+    const votingPeriod = 70
+    const cycle1 = creationHeight + (votingPeriod - creationHeight % votingPeriod) + votingPeriod
+    await testing.container.generate(cycle1 - await testing.container.getBlockCount())
+
+    // Vote on cycle 2
+    const masternodes = await testing.rpc.masternode.listMasternodes()
+    const votes = [VoteDecision.YES, VoteDecision.NO, VoteDecision.NO]
+    let index = 0
+    for (const [id, data] of Object.entries(masternodes)) {
+      if (data.operatorIsMine) {
+        await testing.container.generate(1, data.operatorAuthAddress) // Generate a block to operatorAuthAddress to be allowed to vote on proposal
+        await testing.rpc.governance.voteGov({
+          proposalId: cfpProposalId,
+          masternodeId: id,
+          decision: votes[index]
+        })
+        index++ // all masternodes vote in second cycle
+      }
+    }
+    await testing.container.generate(1)
+  })
+
+  afterAll(async () => {
+    await stopTestingApp(container, app)
+  })
+
+  it('should listProposalVotes', async () => {
+    const result = await controller.listProposalVotes(cfpProposalId)
+    expect(result.data.length).toStrictEqual(3)
```

Comment:
Same here - test the shape on top of the length

Reply:
updated